Mathlib/Analysis/LocallyConvex/Separation.lean

```diff
@@ -316,4 +319,68 @@ theorem iInter_halfSpaces_eq (hs₁ : Convex ℝ s) (hs₂ : IsClosed s) :
   obtain ⟨l, s, hlA, hl⟩ := geometric_hahn_banach_closed_point (𝕜 := 𝕜) hs₁ hs₂ h
   obtain ⟨y, hy, hxy⟩ := hx l
   exact ((hxy.trans_lt (hlA y hy)).trans hl).false
+
+/-- A variant of `iInter_halfSpaces_eq`. If `s` is nonempty, then all the halfspaces are
+nontrivial. -/
+theorem iInter_halfSpaces_const_eq (hs₁ : Convex ℝ s) (hs₂ : IsClosed s) :
+    ∃ (L : (sᶜ : Set E) → StrongDual 𝕜 E) (c : (sᶜ : Set E) → ℝ),
+    ⋂ y, { x | re (L y x) ≤ c y } = s ∧ (s.Nonempty → ∀ y, ∃ x, re (L y x) ≠ 0) := by
+  have (y : (sᶜ : Set E)) := geometric_hahn_banach_closed_point (𝕜 := 𝕜) hs₁ hs₂ y.2
+  choose L c hLc using this
+  refine ⟨L, c, ?_, fun h y => ?_⟩
+  · rw [iInter_setOf]
+    refine subset_antisymm (fun x hx => ?_) fun x hx l => ((hLc l).1 x hx).le
+    by_contra!
+    simp only [Subtype.forall, mem_compl_iff, mem_setOf_eq] at hx
+    linarith [(hLc ⟨x, this⟩).2, hx x this]
+  · by_contra! p
+    have := lt_trans ((hLc y).1 h.some h.some_mem) (hLc y).2
+    simp [p] at this
+
+/-- A closed convex set with a Lindelöf complement is an intersection of countably many
+halfspaces. -/
+theorem _root_.IsLindelof.iInter_countable_halfSpaces_const_eq (hs₁ : Convex ℝ s) (hs₂ : IsClosed s)
+    (hs₃ : IsLindelof sᶜ) : ∃ (u : Set (sᶜ : Set E)) (L : u → StrongDual 𝕜 E) (c : u → ℝ),
+    u.Countable ∧ ⋂ y, { x | re (L y x) ≤ c y } = s ∧
+    (s.Nonempty → ∀ y, ∃ x, re (L y x) ≠ 0) := by
```

Comment:
I think `IsLindelof sᶜ` is too technical as an hypothesis. I suggest assuming `[HereditarilyLindelofSpace E]` instead, which can be proven by typeclass inference without users having to think about what it means.